torch/onnx/symbolic_opset9.py

```diff
@@ -667,59 +667,34 @@ def replication_pad(g, input, padding):
-        width_scale = float(output_size[-1]) / input.type().sizes()[-1]
-        scales = g.op("Constant", value_t=torch.tensor([1., 1., height_scale, width_scale]))
-
-    return g.op("Upsample", input, scales, mode_s="nearest")
-
-
-def upsample_bilinear2d(g, input, output_size, align_corners):
-    align_corners = sym_help._maybe_get_scalar(align_corners)
-    if align_corners:
-        return _unimplemented("upsample_bilinear2d", "align_corners == True")
-
-    output_size = sym_help._maybe_get_const(output_size, 'is')
-    if sym_help._is_value(output_size):
-        offset = 2
-        input_length = len(input.type().sizes())
-        offsets = g.op("Constant", value_t=torch.tensor([1. for i in range(offset)]))
-        dividend = g.op("Cast", output_size, to_i=sym_help.cast_pytorch_to_onnx["Float"])
-        divisor = g.op(
-            "Slice",
-            g.op("Shape", input),
-            axes_i=[0],
-            ends_i=[input_length],
-            starts_i=[offset]
-        )
-        divisor = g.op("Cast", divisor, to_i=sym_help.cast_pytorch_to_onnx["Float"])
-        scale_dims = g.op("Div", dividend, divisor)
-        scales = g.op("Concat", offsets, scale_dims, axis_i=0)
-    else:
-        height_scale = float(output_size[-2]) / input.type().sizes()[-2]
-        width_scale = float(output_size[-1]) / input.type().sizes()[-1]
-        scales = g.op("Constant", value_t=torch.tensor([1., 1., height_scale,
-                                                        width_scale]))
-    return g.op("Upsample", input, scales,
-                mode_s="linear")
+def _interpolate(name, dim, interpolate_mode):
+    def symbolic_fn(g, input, output_size, align_corners=None):
+        align_corners = sym_help._maybe_get_scalar(align_corners)
+        if align_corners:
+            return _unimplemented(name, "align_corners == True")
+
+        output_size = sym_help._maybe_get_const(output_size, 'is')
+        if sym_help._is_value(output_size):
+            offset = 2
+            offsets = g.op("Constant", value_t=torch.tensor([1. for i in range(offset)]))
+            dividend = g.op("Cast", output_size, to_i=sym_help.cast_pytorch_to_onnx["Float"])
+            divisor = _slice_op(g, g.op("Shape", input), axes=[0], ends=[dim], starts=[offset])
+            divisor = g.op("Cast", divisor, to_i=sym_help.cast_pytorch_to_onnx["Float"])
+            scale_dims = g.op("Div", dividend, divisor)
+            scales = g.op("Concat", offsets, scale_dims, axis_i=0)
+        else:
+            scales_constant = [1. if i < 2 else
+                               float(output_size[-(dim - i)]) / float(input.type().sizes()[-(dim - i)])
+                               for i in range(0, dim)]
+            scales = g.op("Constant", value_t=torch.tensor(scales_constant))
+        return g.op("Upsample", input, scales, mode_s=interpolate_mode)
+    return symbolic_fn
+
+
+upsample_nearest1d = _interpolate('upsample_nearest1d', 3, "nearest")
```

Comment:
fuse upsamples together is great!

maybe we should have some convention on the prefix of the helper functions, such as _interpolate, slice_op, they are not real symbolics, just some helper functions.

Reply:
I renamed slice to follow the same convention as _interpolate and other ops (_max_pool, _avg_pool).